libraries/AP_GPS/AP_GPS_SBF.cpp

```diff
@@ -121,6 +122,24 @@ AP_GPS_SBF::read(void)
                                     config_string = nullptr;
                                 }
                                 break;
+                            case Config_State::SBAS:
+                                switch ((AP_GPS::SBAS_Mode)gps._sbas_mode) {
+                                    case AP_GPS::SBAS_Mode::Disabled:
+                                        if (asprintf(&config_string,"%s\n", (char *)sbas_off) == -1) {
+                                            config_string = nullptr;
+                                        }
+                                        break;
+                                    case AP_GPS::SBAS_Mode::Enabled:
+                                        if (asprintf(&config_string,"%s\n", (char *)sbas_on_blob[_init_blob_index]) == -1) {
```

Comment:
On the face of it, this looks like a horrible cast - you're making an immutable value mutable.

I'm also not sure why you're making the cast in any case - removing it still seems to allow things to compile for me.
```suggestion
                                        if (asprintf(&config_string, "%s\n", (char *)sbas_on_blob[_init_blob_index]) == -1) {
```

Reply:
I'll be honest, I can't figure out why either. I think it was needed at one point, but it's building fine locally for SITL (GCC 10.2.0) and ChibiOS (gcc 6.3.1) so seems like it can be removed.